internal/serve/httperror/errors.go

```diff
@@ -8,61 +8,65 @@ import (
 	"github.com/stellar/go/support/render/httpjson"
 )
 
-type errorResponse struct {
-	Status int    `json:"-"`
-	Error  string `json:"error"`
+type ErrorResponse struct {
+	Status int                    `json:"-"`
+	Error  string                 `json:"error"`
+	Extras map[string]interface{} `json:"extras,omitempty"`
 }
 
-func (e errorResponse) Render(w http.ResponseWriter) {
+func (e ErrorResponse) Render(w http.ResponseWriter) {
 	httpjson.RenderStatus(w, e.Status, e, httpjson.JSON)
 }
 
 type ErrorHandler struct {
-	Error errorResponse
+	Error ErrorResponse
 }
 
 func (h ErrorHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	h.Error.Render(w)
 }
```

Comment:
Is the error handler necessary? I don't know the rest of the code, so I'm trying understand if this abstraction is needed.

Reply:
Thank you for the question! It's used to override the default messages for some HTTP statuses. Please refer to this:

https://github.com/stellar/wallet-backend/blob/main/internal/serve/serve.go#L79-L80